modules/gapi/samples/face_beautification.cpp

```diff
@@ -0,0 +1,796 @@
+using Landmarks = std::vector<cv::Point>;
+
+
+// Wrapper function
+template<typename Tp> inline int toIntRounded(const Tp x)
+{
+    return static_cast<int>(std::lround(x));
+}
+
+template<typename Tp> inline double toDouble(const Tp x)
+{
+    return static_cast<double>(x);
+}
+
+std::string getWeightsPath(const std::string &mdlXMLPath) // mdlXMLPath =
+                                                          // "The/Full/Path.xml"
+{
+    size_t size = mdlXMLPath.size();
+    CV_Assert(mdlXMLPath.substr(size - 4, size)           // The last 4 symbols
+                  == ".xml");                             // must be ".xml"
+    std::string mdlBinPath(mdlXMLPath);
+    return mdlBinPath.replace(size - 3, 3, "bin");        // return
+                                                          // "The/Full/Path.bin"
+}
+} // anonymous namespace
+
+
+
+namespace custom
+{
+using TplPtsFaceElements_Jaw = std::tuple<cv::GArray<Landmarks>,
+                                          cv::GArray<Contour>>;
+using TplFaces_FaceElements  = std::tuple<cv::GArray<Contour>,
+                                          cv::GArray<Contour>>;
+
+// Wrapper-functions
+inline int getLineInclinationAngleDegrees(const cv::Point &ptLeft,
+                                          const cv::Point &ptRight);
+inline Contour getForeheadEllipse(const cv::Point &ptJawLeft,
+                                  const cv::Point &ptJawRight,
+                                  const cv::Point &ptJawMiddle,
+                                  const size_t     capacity);
+inline Contour getEyeEllipse(const cv::Point &ptLeft,
+                             const cv::Point &ptRight,
+                             const size_t     capacity);
+inline Contour getPatchedEllipse(const cv::Point &ptLeft,
+                                 const cv::Point &ptRight,
+                                 const cv::Point &ptUp,
+                                 const cv::Point &ptDown);
+
+// Networks
+G_API_NET(FaceDetector,  <cv::GMat(cv::GMat)>, "face_detector");
+G_API_NET(LandmDetector, <cv::GMat(cv::GMat)>, "landm_detector");
+
+// Function kernels
+G_TYPED_KERNEL(GBilatFilter, <cv::GMat(cv::GMat,int,double,double)>,
+               "custom.faceb12n.bilateralFilter")
+{
+    static cv::GMatDesc outMeta(cv::GMatDesc in, int,double,double)
+    {
+        return in;
+    }
+};
+
+G_TYPED_KERNEL(GLaplacian, <cv::GMat(cv::GMat,int)>,
+               "custom.faceb12n.Laplacian")
+{
+    static cv::GMatDesc outMeta(cv::GMatDesc in, int)
+    {
+        return in;
+    }
+};
+
+G_TYPED_KERNEL(GFillPolyGContours, <cv::GMat(cv::GMat,cv::GArray<Contour>)>,
+               "custom.faceb12n.fillPolyGContours")
+{
+    static cv::GMatDesc outMeta(cv::GMatDesc in, cv::GArrayDesc)
+    {
+        return in.withType(CV_8U, 1);
+    }
+};
+
+G_TYPED_KERNEL(GPolyLines, <cv::GMat(cv::GMat,cv::GArray<Contour>,bool,
+                                     cv::Scalar)>,
+               "custom.faceb12n.polyLines")
```

Comment:
The formatting still looks weird. I think I will align it on my own once it merged.

Reply:
OK. I figured out how to align it better, I think..